Core/include/Acts/TrackFitting/GainMatrixUpdater.hpp

```diff
@@ -81,8 +80,17 @@ class GainMatrixUpdater {
             trackState.predictedCovariance(),
             trackState.filtered(),
             trackState.filteredCovariance(),
-            trackState.calibrated(),
-            trackState.calibratedCovariance(),
+            // This abuses an incorrectly sized vector / matrix to access the
+            // data pointer! This works (don't use the matrix as is!), but be
+            // careful!
+            trackState
```

Comment:
Would it possible to hide this c++ syntax mess somehow in the TracksState?

Reply:
Not easily. We could explicitly expose the raw `const double*` / `double*` to allow construction of custom proxy types, but I wasn't sure that's a great idea either.